src/include/parser/prepare_statement.h

```diff
@@ -29,8 +29,8 @@ class PrepareStatement : public SQLStatement {
    * @param query - the parsed form of statement
    * @param placeholders - placeholder values? (explain)
    */
-  PrepareStatement(std::string name, std::shared_ptr<SQLStatement> query,
-                   std::vector<std::shared_ptr<ParameterValueExpression>> placeholders)
+  PrepareStatement(std::string name, std::unique_ptr<SQLStatement> query,
+                   std::vector<common::ManagedPointer<ParameterValueExpression>> placeholders)
```

Comment:
i don't see placeholders being assigned. I think you're missing `placeholders_(placeholders)`

Reply:
https://github.com/cmu-db/terrier/blob/5d983d30e42dcd52c11e07c36a6141ded2acbae5/src/include/parser/prepare_statement.h#L37 ?